app/javascript/components/namespace-form/index.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './namespace-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (isDomain, namespaceId, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_${isDomain ? 'domains' : 'namespaces'}/${namespaceId}?attributes=name,description,enabled,source`).then((result) => {
+    miqSparkleOff();
+    f.setInitialValues({ name: result.name, description: result.description, enabled: result.enabled });
+    f.setDisabled(['user', 'user_locked'].includes(result.source));
```

Comment:
You should put a `FIXME` here to retrieve the readability of the record using the API directly instead of manually testing in the frontend. cc @tinaafitz

Reply:
@skateman is this happening because the UI treats namespaces as the top level object in certain places and domain as the top level object at certain places?

Reply:
No, this is a much simpler problem. Right now we can't ask the API if these fields are writable. We checked how it has been done in the UI and hardcoded it here, but it's a bad pattern to have such logic in the frontend code.